tools/src/main/java/org/apache/kafka/tools/LogDirsCommand.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.tools;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import net.sourceforge.argparse4j.ArgumentParsers;
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.Namespace;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.admin.LogDirDescription;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.utils.Exit;
+import org.apache.kafka.common.utils.Utils;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+import static net.sourceforge.argparse4j.impl.Arguments.store;
+
+public class LogDirsCommand {
+    public static void main(String... args) {
+        Exit.exit(mainNoExit(args));
+    }
+
+    static int mainNoExit(String... args) {
+        try {
+            execute(args);
+            return 0;
+        } catch (Throwable e) {
+            System.err.println(e.getMessage());
+            System.err.println(Utils.stackTrace(e));
+            return 1;
+        }
+    }
+
+    static void execute(String[] args) throws ExecutionException, InterruptedException, IOException {
+        ArgumentParser parser = ArgumentParsers
```

Comment:
To be honest, I used `argparse4j` because I looked at https://github.com/apache/kafka/pull/13080/files as a reference. I don't have a strong opinion either way, but the first output you presented seems nicer to me.

Reply:
We should not change the argument parsing library in this PR. This is likely to slightly change the usage and could potentially break existing commands/scripts used by users. Yes ideally we should use a single library but we can do that after (and it's likely to require a KIP).

Reply:
Sorry, then may I get a bit of clarification on how do you envision this to work? Do you envision that a Java compatible copy of `CommandDefaultOptions` is written? I am asking this because I was battling with moving ConsoleConsumer and there I ran into the following problem:
```
abstract class CommandDefaultOptions(val args: Array[String], allowCommandOptionAbbreviation: Boolean = false) {
  val parser = new OptionParser(allowCommandOptionAbbreviation)
...
```
```
private static class ConsumerConfig extends CommandDefaultOptions {
        ConsumerConfig(String... args) {
            super(args, false)
            
            ... = this.parser(); <--- Cannot access joptsimple.OptionParser
```
Is it in general that we are only trying to move the commands from Scala to Java or do we want a complete break from Scala classes?

Reply:
> If we are completely moving to Java does this mean that CommandLineUtils needs to be rewritten first?

Yes, I already created `CommandLineUtils` as part of the `DumpLogSegments` migration, which also includes `CommandDefaultOptions`, but I still have some work to do before opening the PR. Let me know if you want a separate PR with just these 3 shared classes.

That said, I think we will need to have tools->core temporary dependency to avoid duplicating critical code like `GroupMetadataManager.formatRecordKeyAndValue(record)` and `TransactionLog.formatRecordKeyAndValue(record)`.

Reply:
If it isn't too big of a problem may I have the 2 classes and see how far I can go with them at least in this pull request?

Reply:
Yes, I'll open a dedicated PR.

Reply:
I completed the migration of these shared classes, but I still need to do some testing before opening the PR (probably tomorrow).